testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -69,11 +73,11 @@ use stacks_common::util::hash::{bytes_to_hex, Sha512Trunc256Sum};
 use stacks_common::util::secp256k1::MessageSignature;
 use url::Url;
 
-use super::config::{EventKeyType, EventObserverConfig};
-
 #[cfg(any(test, feature = "testing"))]
-pub static TEST_SKIP_BLOCK_ANNOUNCEMENT: std::sync::Mutex<Option<bool>> =
-    std::sync::Mutex::new(None);
+lazy_static! {
+    /// Do not announce a signed/mined block to the network when set to true.
+    pub static ref TEST_SKIP_BLOCK_ANNOUNCEMENT: TestFlag<bool> = TestFlag::default();
+}
```

Comment:
Nit: I think we want to move to std::sync::LazyLock / LazyCell https://github.com/stacks-network/stacks-core/issues/5035

Reply:
I will leave it for now but will keep in mind for a follow on. Just don't want to hold this up any further XD